src/main/java/ch/njol/skript/effects/EffEquip.java

```diff
@@ -91,85 +105,80 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
 						if (CARPET.isOfType(item)) {
-							invi.setDecor(item);
+							inv.setDecor(equip ? item : AIR);
 						} else if (CHEST.isOfType(item)) {
-							((Llama) en).setCarryingChest(true);
+							((Llama) en).setCarryingChest(equip);
 						}
 					}
 				}
 				continue;
-			} else if (NEW_HORSES && en instanceof AbstractHorse) {
+			} else if (en instanceof AbstractHorse) {
 				// Spigot's API is bad, just bad... Abstract horse doesn't have horse inventory!
-				Inventory invi = ((AbstractHorse) en).getInventory();
+				Inventory inv = ((AbstractHorse) en).getInventory();
 				for (ItemType t : ts) {
 					for (ItemStack item : t.getAll()) {
 						if (SADDLE.isOfType(item)) {
-							invi.setItem(0, item); // Slot 0=saddle
+							inv.setItem(0, equip ? item : AIR); // Slot 0=saddle
 						} else if (HORSE_ARMOR.isOfType(item)) {
-							invi.setItem(1, item); // Slot 1=armor
+							inv.setItem(1, equip ? item : AIR); // Slot 1=armor
 						} else if (CHEST.isOfType(item) && en instanceof ChestedHorse) {
-							((ChestedHorse) en).setCarryingChest(true);
-						}
-					}
-				}
-				continue;
-			} else if (SUPPORTS_HORSES && en instanceof Horse) {
-				HorseInventory invi = ((Horse) en).getInventory();
-				for (ItemType t : ts) {
-					for (ItemStack item : t.getAll()) {
-						if (SADDLE.isOfType(item)) {
-							invi.setSaddle(item);
-						} else if (HORSE_ARMOR.isOfType(item)) {
-							invi.setArmor(item);
-						} else if (CHEST.isOfType(item)) {
-							((Horse) en).setCarryingChest(true);
+							((ChestedHorse) en).setCarryingChest(equip);
 						}
 					}
 				}
 				continue;
 			}
-			EntityEquipment equip = en.getEquipment();
-			if (equip == null)
+			EntityEquipment equipment = en.getEquipment();
+			if (equipment == null)
 				continue;
 			for (ItemType t : ts) {
 				for (ItemStack item : t.getAll()) {
-					// Blocks are visible in head slot, too
-					// TODO skulls; waiting for decoration aliases
-					if (HELMET.isOfType(item) || item.getType().isBlock())
-						equip.setHelmet(item);
-					else if (CHESTPLATE.isOfType(item))
-						equip.setChestplate(item);
-					else if (LEGGINGS.isOfType(item))
-						equip.setLeggings(item);
-					else if (BOOTS.isOfType(item))
-						equip.setBoots(item);
-					
-					// We have no idea where to equip other items
-					// User can set them to slot they need custom hats etc.
+					if (CHESTPLATE.isOfType(item)) {
+						equipment.setChestplate(equip ? item : AIR);
+					} else if (LEGGINGS.isOfType(item)) {
+						equipment.setLeggings(equip ? item : AIR);
+					} else if (BOOTS.isOfType(item)) {
+						equipment.setBoots(equip ? item : AIR);
+					} else {
+						// Apply all other items to head, as all items will appear on a player's head
+						equipment.setHelmet(equip ? item : AIR);
+					}
+				}
+				if (unequipHelmet) { // Since players can wear any helmet, ts won't have the item in the array every time
+					equipment.setHelmet(AIR);
 				}
```

Comment:
```suggestion
				if (unequipHelmet) // Since players can wear any helmet, ts won't have the item in the array every time
					equipment.setHelmet(AIR);
```